### PR TITLE
Lightgcn fix

### PR DESCRIPTION
**Pull request by theisjendal**
### Description
LightGCN is a simplified version of NGCF, however, some of the simplifications made in LightGCN where not included.
Therefore: 

* I have removed normalization between layers as this is not used by LightGCN.
* Fixed the sum constant used for adding layers to `1/(L+1)` and instead of `1/(l+1)` where `L` is the number of layers and `l` is the current layer index.
* Previous version of LightGCN didn't support blocks. It now supports DGL blocks.
* Fixed requirements versions as newer version of DGL/Torch created an error. 
* I moved edge normalization to graph creation making computations in LightGCN easier/simpler.

### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

**Comment by theisjendal** [comment]
@tqtg as per 
https://github.com/PreferredAI/cornac/pull/600#issuecomment-1995827405, I've created a pull request for LightGCN only.

File: cornac/models/lightgcn/lightgcn.py [hunk 1]
```diff
@@ -92,16 +82,33 @@ def __init__(self, g, in_size, num_layers, lambda_reg, device=None):
...
+            # obtain features of each layer and concatenate them all
+            user_embeds = h_dict[USER_KEY]
+            item_embeds = h_dict[ITEM_KEY]
+            iterator = enumerate(zip([in_g] * len(self.layers), self.layers))
+
+        user_embeds = user_embeds * (1 / (len(self.layers) + 1))
+        item_embeds = item_embeds * (1 / (len(self.layers) + 1))
 
-        for k, layer in enumerate(self.layers):
+        for k, (g, layer) in iterator:
```
Comment (tqtg) [inline cornac/models/lightgcn/lightgcn.py:102]: Since the weights are uniform, we can simplify the whole logic here by summing up all the embeddings and divide by `(len(self.layers) + 1)` at the end.

**Comment by tqtg** [comment]
thanks @theisjendal for clarifying all the changes. They all make sense to me after revising the [authors' implementation](https://github.com/gusye1234/LightGCN-PyTorch/blob/master/code/model.py) of LightGCN. Could you also try to compare the updated version and the old one using [this example](https://github.com/PreferredAI/cornac/blob/master/examples/lightgcn_example.py) with the same hyper-parameters? I'm curious about the changes in accuracy and speed of convergence.

File: cornac/models/lightgcn/lightgcn.py (hunk 1 above)
Comment (theisjendal) [reply cornac/models/lightgcn/lightgcn.py:102]: Did some refactoring in commit 70eb292

**Comment by theisjendal** [comment]
OLD
Training:  11%, 110/1000 [13:45<1:50:48,  7.47s/iter, loss=0.00479]
Early stopping:                                                                                                                                                                                                                                             
- best epoch = 61, stopped epoch = 111
- best monitored value = 0.247716 (delta = -0.005755)


NEW
Training:  62%, 619/1000 [1:11:29<44:35,  7.02s/iter, loss=0.00469]
Early stopping:                                                                                                                                                                                                                                             
- best epoch = 570, stopped epoch = 620
- best monitored value = 0.244275 (delta = -0.001430)
                              


VALIDATION:
```
            | NDCG@20 | Recall@20 | Time (s)
------------ + ------- + --------- + --------
LightGCN-Old |  0.1534 |    0.2420 |   4.9250
LightGCN-New |  0.1552 |    0.2428 |   4.8821

```
TEST:
```
             | NDCG@20 | Recall@20 | Train (s) | Test (s)
------------ + ------- + --------- + --------- + --------
LightGCN-Old |  0.1698 |    0.2503 |  831.2188 |   5.2217
LightGCN-New |  0.1717 |    0.2581 | 4294.9170 |   5.2097

```

**Comment by tqtg** [comment]
LGTM. Feel free to merge this PR.

**Review by tqtg — APPROVED** [review]
